specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -160,40 +159,41 @@ export const businessRuleDefs: MappedBusinessRuleDefs = {
     },
     fieldChecks: {
       collectionObjectType: (resource) => {
-        // Fields ignored for better UX/backend restrictions
-        const fieldsToIgnore = [
-          'cataloger',
-          'catalogNumber',
-          'collection',
-          'collectionObjectType',
-          'version',
-        ];
-
-        // Clear all fields of CO except to-many dependents
-        resource.specifyTable.fields
-          .filter((field) => !f.includes(fieldsToIgnore, field.name))
-          .map((field) => {
-            const fieldName = field.name as keyof (CollectionObject['fields'] &
-              CollectionObject['toManyDependent'] &
-              CollectionObject['toManyIndependent'] &
-              CollectionObject['toOneDependent'] &
-              CollectionObject['toOneIndependent']);
-            if (
-              !field.isRelationship ||
-              !relationshipIsToMany(field) ||
-              !field.isDependent()
-            )
-              resource.set(fieldName, null);
-          });
-
         // Delete all determinations
         const determinations = resource.getDependentResource('determinations');
-        while (
-          determinations !== undefined &&
-          determinations.models.length > 0
-        ) {
-          determinations.remove(determinations.models[0]);
+        if (determinations?.models.length === 0) {
+          return;
         }
+        const taxon = determinations?.models[0].get('taxon');
+        if (typeof taxon !== 'string') return;
+        const taxonId = idFromUrl(taxon ?? '');
+        const COType = resource.get('collectionObjectType');
+        const COTypeId = idFromUrl(COType);
+
+        const fetchTaxon = fetchResource('Taxon', taxonId ?? 0);
+        const fetchCOType = fetchResource(
+          'CollectionObjectType',
+          COTypeId ?? 0
```

Comment:
For taxonId ?? 0  and COTypeId ?? 0 , we can just handle the case when they are null or undefined by returning.